README.md
# get_code_status

Console color display of all http status codes. For those who never remember http codes.

## Installation

```bash
pub add get_code_status
```

## Usage

```dart
import 'package:get_code_status/get_code_status.dart';

void main() {
  var getStatus = GetCodeStatus();
  //print colorful code in console.
  getStatus.printCodeStatus(500);
  //get code status as String
  print(getStatus.codeStatusToString(500).notes);
}
```


CHANGELOG.md
## 1.0.0

- Initial version.
## 1.0.1

- Add "toString" function.

test/get_code_status_test.dart
import 'package:get_code_status/src/get_code.dart';
import 'package:test/test.dart';

void main() {
  group('A group of tests', () {
    final getStatus = GetCodeStatus();

    setUp(() {
      // Additional setup goes here.
    });

    test('First Test', () {
      expect(getStatus, isTrue);
    });
  });
}


lib/src/get_code_status_base.dart
part of 'get_code.dart';

class GetCodeStatus {
  DataCode data = DataCode();
  void printCodeStatus(int code) {
    PrintColor printColor = PrintColor();

//-----------------------------------------------------------------------------
    final index =
        data.getFromJson().indexWhere((element) => element.code == code);

    if (!index.isNegative) {
      var stat = data.getFromJson().elementAt(index);

      String message = "[${stat.code}] [${stat.status}], ${stat.notes}";

      if (code == 101 || code == 100) {
        print(message);
      } else if (code >= 200 && code <= 206) {
        printColor.successful(message);
      } else if (code >= 300 && code <= 307) {
        printColor.redirection(message);
      } else if (code >= 400 && code <= 417) {
        printColor.clientError(message);
      } else if (code >= 500 && code <= 505) {
        printColor.serverError(message);
      }
    } else {
      print('\x1B[36m Unknow status code \x1B[0m');
    }
  }

  StatusCodeModel codeStatusToString(int code) {
    final index =
        data.getFromJson().indexWhere((element) => element.code == code);

    if (!index.isNegative) {
      var stat = data.getFromJson().elementAt(index);

      return stat;
    } else {
      return data.getFromJson().elementAt(0);
    }
  }
}


lib/src/data/data.dart
part of '../get_code.dart';

class Data {
  String get getData => _data;

  final String _data = """
[
    {
        "code": 0,
        "status":"Unknown",
        "notes":"Unknown"
     
    },
    {
        "code": 100,
        "status":"",
        "notes":"Continue"
     
    },
    {
        "code": 101,
        "status":"",
        "notes":"Switching Protocols"
    },
    {
        "code": 200,
        "status":"Successful",
        "notes": "OK : Everything is normal"
    },
    {
        "code": 201,
        "status":"Successful",
        "notes": "Created"
    },
    {
        "code": 202,
        "status":"Successful",
        "notes": "Accepted"
    },
    {
        "code": 203,
        "status":"Successful",
        "notes": "Non-Authoritative Information"
    },
    {
        "code": 204,
        "status":"Successful",
        "notes": "No Content"
    },
    {
        "code": 205,
        "status":"Successful",
        "notes": "Reset Content"
    }
    ,{
        "code": 206,
        "status":"Successful",
        "notes": "Partial Content"
    },{
        "code": 300,
        "status":"Redirection",
        "notes": "Multiple Choices"
    },{
        "code": 301,
        "status":"Redirection",
        "notes": "Moved Permanently : Update your URL, this has moved for good."
    },{
        "code": 302,
        "status":"Redirection",
        "notes": "Found"
    },{
        "code": 303,
        "status":"Redirection",
        "notes": "See Other"
    },{
        "code": 304,
        "status":"Redirection",
        "notes": "Not Modified"
    },{
        "code": 305,
        "status":"Redirection",
        "notes": "Use Proxy"
    },{
        "code": 306,
        "status":"Redirection",
        "notes": "Unused"
    },{
        "code": 307,
        "status":"Redirection",
        "notes": "Temporary Redirect : This is temporarly moved, don't update your bookmarks."
    },{
        "code": 400,
        "status":"Client Error",
        "notes": "Bad Request : Server didn't understand the URL you gave it."
    },{
        "code": 401,
        "status":"Client Error",
        "notes": "Unauthorized : Must be authenticated."
    },{
        "code": 402,
        "status":"Client Error",
        "notes": "Payment Required : Not used really"
    },{
        "code": 403,
        "status":"Client Error",
        "notes": "Forbidden : Server refuses to give you a file, authentication won't help"
    },{
        "code": 404,
        "status":"Client Error",
        "notes": "Not Found : A file doesn't exist at that address."
    },{
        "code": 405,
        "status":"Client Error",
        "notes": "Method Not Allowed"
    },{
        "code": 406,
        "status":"Client Error",
        "notes": "Not Acceptable"
    },{
        "code": 407,
        "status":"Client Error",
        "notes": " "
    },{
        "code": 408,
        "status":"Client Error",
        "notes": "Request Timeout : Browser took too long to request something."
    },{
        "code": 409,
        "status":"Client Error",
        "notes": "Conflict"
    },{
        "code": 410,
        "status":"Client Error",
        "notes": "Gone"
    },{
        "code": 411,
        "status":"Client Error",
        "notes": "Lengh Required"
    },{
        "code": 412,
        "status":"Client Error",
        "notes": "Precondition Failed"
    },{
        "code": 413,
        "status":"Client Error",
        "notes": "Reqeust Entity Too Large"
    },{
        "code": 415,
        "status":"Client Error",
        "notes": "Unsupported Media Type"
    },{
        "code": 416,
        "status":"Client Error",
        "notes": "Request Range Not Satisfiable"
    }
    ,{
        "code": 417,
        "status":"Client Error",
        "notes": "Expectation Failed"
    }
    ,{
        "code": 500,
        "status":"Server Error",
        "notes": "Internal Server Error : Something on the server didn't work right."
    },{
        "code": 501,
        "status":"Server Error",
        "notes": "Not Implemented"
    },{
        "code": 502,
        "status":"Server Error",
        "notes": "Bad Gateway"
    },{
        "code": 503,
        "status":"Server Error",
        "notes": "Service Unavailable : Too busy to respond to a client."
    },{
        "code": 504,
        "status":"Server Error",
        "notes": "Gateway Timeout"
    },{
        "code": 505,
        "status":"Server Error",
        "notes": "HTTP Version Not Supported"
    }
]
""";
}
